### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by jdkandersson**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -4,16 +4,113 @@ on:
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/build_oci_image
+        id: build
+        with:
+          dockerfile: jenkins-agent.Dockerfile
+          resource-name: jenkins-agent-image
+  integration-test-lxd-microk8s:
+    name: Integration tests (lxd and microk8s)
+    # [20220824] charmed-kubernetes/actions-operator@main does not work on
```
Comment (merkata) [inline .github/workflows/ci.yaml:41]: Good comment, I was looking at the latest commit only and wanted to suggest using the actions-operator - this tells me why it's not used.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -4,16 +4,113 @@ on:
...
+        uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: lxd
+      - name: Install microk8s
+        run: |
+          sudo snap install microk8s --classic
+          sudo microk8s status --wait-ready
+          sudo microk8s.enable dns hostpath-storage
```
Comment (merkata) [inline .github/workflows/ci.yaml:56]: Could we enable the registry as well, do we need the additional step bellow?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -4,16 +4,113 @@ on:
...
+          sudo usermod -a -G microk8s $USER
+          sudo chown -f -R $USER ~/.kube
+      - name: Enable microk8s registry
+        run: |
+          sudo microk8s.enable registry
+      - name: Setup jenkins
+        id: jenkins
+        run: |
+          echo Installing selinium for Jenkins configuration
```
Comment (merkata) [inline .github/workflows/ci.yaml:65]: typo selenium

**Review by merkata — COMMENTED** [review]
I know it's still WIP, just wanted to say it looks really good and the comments are helpful in navigating a big sized PR like this one. Good stuff!

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (jdkandersson) [reply .github/workflows/ci.yaml:56]: Thank you, fixed

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (jdkandersson) [reply .github/workflows/ci.yaml:65]: Thank you, fixed

**Comment by jdkandersson** [comment]
Closing and re-opening for GitHub to correctly show diff to main

**Pull request closed without merging**